applications/mlflow/kots/kots-config.yaml

```diff
@@ -0,0 +1,200 @@
+---
+apiVersion: kots.io/v1beta1
+kind: Config
+metadata:
+  name: config
+spec:
+  groups:
+  # Ingress settings
+  - name: ingress_settings
+    title: Ingress Settings
+    description: Configure Ingress for Mlflow
+    items:
+    - name: mlflow_ingress_type
+      title: Mlflow Ingress Type
+      help_text: | 
+        Select how traffic will ingress to the Mlflow appliction. The Ingress Controller option will create an Ingress object, 
+        and Load Balancer will configure Mlflow's Kubernetes service to be of type LoadBalancer.
+      type: select_one
+      items:
+      - name: ingress_controller
+        title: Ingress Controller
+      - name: load_balancer
+        title: Load Balancer
+      default: "ingress_controller"
+      required: true
+      when: 'repl{{ ne Distribution "embedded-cluster" }}'
+    - name: mlflow_ingress_host
+      title: Mlflow Hostname
+      help_text: Hostname which will be used to access Mlflow
+      type: text
+      default: "mlflow.example.com"
+      required: true
+      when: 'repl{{ ConfigOptionEquals "mlflow_ingress_type" "ingress_controller" }}'
+    - name: mlflow_ingress_class_name
+      type: text
+      title: Ingress Class Name
+      default: "nginx"
+      when: 'repl{{ and (ne Distribution "embedded-cluster") (ConfigOptionEquals "mlflow_ingress_type" "ingress_controller") }}'
```

Comment:
^ Question about the `when` property in a few of these config items

It looks like `mlflow_ingress_type` only shows up when the cluster is _not_ an embedded cluster

Based on that, I was wondering why `(ne Distribution "embedded-cluster")` is required in the `when` statement for the `mlflow_ingress_class_name` item. My initial thought was that, if "mlflow_ingress_type" is set to "ingress_controller", then that automatically means that it's not an embedded cluster install.

Is it because `mlflow_ingress_type` always default to `ingress_controller`, even if the cluster is embedded and `mlflow_ingress_type` isn't displayed on the config page?

Reply:
Yeah, i see what you're saying. The intention here is:
* If its embedded cluster, the only ingress config shown is for setting the hostname as well as the TLS settings

So I need to test to make sure that if `mlflow_ingress_type` has its `when` conditional evaluating to `false`, despite its default being  `ingress_controller` that `(ConfigOptionEquals "mlflow_ingress_type" "ingress_controller")` still evaluates to `false`. If that's the case, then yeah I agree we can remove the embedded cluster part from the conditional and simplify this.

That make sense?

Reply:
Ah okay, I think i finally understood the confusion here. 

> My initial thought was that, if "mlflow_ingress_type" is set to "ingress_controller", then that automatically means that it's not an embedded cluster install.

So the distinction here isn't that it won't be an embedded cluster installer if `mlflow_ingress_type` is set to `ingress_controller` its that in Embedded cluster the ingress type should always be `ingress_controller`, so i don't want the user to change it. 

The `(ne Distribution "embedded-cluster")` is use to further filter out ingress config options that i don't think a user will need to set in embedded cluster installs.

Reply:
Ahh okay, got it. Makes sense. And nice I think comments are a great idea!